packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -67,7 +166,30 @@ async function createNoteFromTemplate(): Promise<void> {
     return;
   }
 
-  const defaultFileName = 'new-note.md';
+  const templateText = await workspace.fs.readFile(
+    URI.joinPath(templatesDir, selectedTemplate)
+  );
+
+  const givenValues = new Map<string, string>();
+  const variables = findFoamVariables(templateText.toString());
+
+  // TODO: Always ask for FOAM_TITLE_SLUG?
```

Comment:
I take this is related to the open question in the PR.
if the template uses `FOAM_TITLE` or `FOAM_TITLE_SLUG`, we ask for the title of the note, and use that as base for both variables (that is, we wouldn't prompt for the slug, and we wouldn't prompt for the title if neither it nor its slug is required).
If no default path property is provided via the template, we default the path to current file dir + `New Note.md`, but in that case we always use that only as default (that is, we still prompt the user to confirm the final filepath).

Keep in mind that in Foam we don't resolve the title to its slugified filename, so our default behavior needs to match that so that wikilinks don't need to be slugified.

Reply:
I've left the behaviour as is, for the case where `FOAM_TITLE` is not provided nor requested by the template.

---

> Keep in mind that in Foam we don't resolve the title to its slugified filename, so our default behavior needs to match that so that wikilinks don't need to be slugified.

I had some difficulty parsing this sentence.

Is the concern that we won't be able to derive `FOAM_TITLE_SLUG` from `FOAM_TITLE` in a way that matches the way that the current wikilinks implementation does? In a future where clicking a wikilink passes in the value of `FOAM_TITLE`, we won't end up producing the same filename that would be produced today?

Reply:
What I mean here is that the wikilink `[[This is my title]]` looks for a file called `This is my title.md` (like Obsidian), not `this-is-my-title.md` (like Markdown Notes).

Reply:
Some thoughts....

What is the preferred slug format then? `Obsidian` or `Markdown Notes`?

1. It sound like the `Obsidian` format is preferable, since otherwise it would be a breaking change when we change `createNoteFromPlacehoder` to use `createNoteFromTemplate` (and pass in the title).
    * Alternatively, if the `Markdown Notes` format is preferable, `Foam: Open Resource` could just be changed to look in both locations, which maintains compatibility.  
2. The existing code was already defaulting to `new-note`. Was that a mistake?
3. If there isn't any slugifying work to be done (`Obsidian`), is there still value in having `FOAM_TITLE_SLUG` at all?

Reply:
@riccardoferretti I think we need to answer these before we can merge this? ☝️

Reply:
ooops, totally missed this

The answer is Obsidian.

1. Obsidian
2. the code was created before the migration to the Obsidian format (for historic reasons before we were using the markdown notes semantics)
3. It could still be used, because someone might want to create `A new note` that results in `a-new-note.md` (if the slug is used for the file path), aware that links to this file will look like `[[a-new-note]]`. Possibly an edge case, but that's basically the semantics of Markdown Notes (except for the link part) - thoughts?

Reply:
> Possibly an edge case... - thoughts?

@riccardoferretti If we want Foam to move towards using Obsidian style links everywhere, let's not add edge case support for this other format.

I'll rework this PR a bit to remove any reference to `FOAM_TITLE_SLUG`. I still like the refactor of the code to use a `Resolver`, and I've based my next work on it.

Reply:
https://github.com/foambubble/foam/pull/586